bennu-core/src/main/java/org/fenixedu/bennu/core/domain/User.java

```diff
@@ -267,10 +267,15 @@ public static User findByUsername(final String username) {
     }
 
     private static User manualFind(String username) {
```

Comment:
This method is being used in map.computeIfAbsent which results in a race condition